python/pyiceberg/expressions/visitors.py

```diff
@@ -675,6 +702,57 @@ def visit_less_than_or_equal(self, term: BoundTerm[L], literal: Literal[L]) -> b
 
         return ROWS_MIGHT_MATCH
 
+    def visit_starts_with(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
+        pos = term.ref().accessor.position
+        field = self.partition_fields[pos]
+        prefix = str(literal.value)
+
+        # contains_null encodes whether at least one partition value is null,
+        # lowerBound is null if all partition values are null
+        all_null = field.contains_null is True and field.lower_bound is None
+        if all_null or not field.lower_bound or not field.upper_bound:
+            return ROWS_CANNOT_MATCH
+
+        lower = _from_byte_buffer(term.ref().field.field_type, field.lower_bound)
+        # truncate lower bound so that its length is not greater than the length of prefix
+        if lower and lower[: len(str(prefix))] > prefix:
```

Comment:
```suggestion
        if lower is not None and lower[: len(str(prefix))] > prefix:
```

Should we put `len(str(prefix))` in a variable? `prefix_len`?